app/actions/index.js

```diff
@@ -91,3 +91,12 @@ export const addDat = key => dispatch => {
     }, 1000)
   })
 }
+
+export const deleteDat = key => dispatch => {
+  dispatch({ type: 'REMOVE_DAT', key })
+  const path = `${homedir()}/Downloads/${key}`
+  Dat(path, { key }, (error, dat) => {
```

Comment:
since the `dat` object has been created already, we could optimize this by storing open dats in a map above:

```js
const dats = new Map()
// ...
dats.set(dat.key, dat)
```

and then here:

```js
const dat = dats.get(key)
dat.close()
dats.delete(key)
```

Reply:
@juliangruber  implemented, thanks for tips, really appreciate It ^_^